src/s3/listobjects.cpp

```diff
@@ -56,90 +60,206 @@ asio::awaitable<void> irods::s3::actions::handle_listobjects_v2(
+                    key = key.substr(1);
+                }
+                key += "/";
+                object.put("Prefix", key);
+                document.add_child("ListBucketResult.CommonPrefixes", object);
+            }
+
+            // Get the data objects within the collection
+            query = fmt::format(
+                "select COLL_NAME, DATA_NAME, DATA_OWNER_NAME, DATA_SIZE, DATA_MODIFY_TIME where COLL_NAME = '{}'",
+                full_path.parent_path().string());
+            std::cout << query << std::endl;
+            for (auto&& i : irods::query<RcComm>(thing.get(), query)) {
+                found_objects = true;
+                ptree object;
+                std::string key = (i[0].size() > base_length ? i[0].substr(base_length) : "") + "/" + i[1];
+                if (key.rfind("/", 0) == 0) {
+                    key = key.substr(1);
+                }
+                object.put("Key", key);
+                object.put("Etag", i[0] + i[1]);
+                object.put("Owner", i[2]);
+                object.put("Size", atoi(i[3].c_str()));
+                try {
+                    std::time_t modified_epoch_time = boost::lexical_cast<std::time_t>(i[4]);
+                    std::string modified_time_str = irods::s3::api::common_routines::convert_time_t_to_str(modified_epoch_time, "%Y-%m-%dT%H:%M:%S.000Z");
+                    object.put("LastModified", modified_time_str);
+                } catch ( const boost::bad_lexical_cast& ) {
+                    // do nothing - don't add LastModified tag
+                }
+                document.add_child("ListBucketResult.Contents", object);
+            }
+        } else {
+
+            // Path does not end in a slash.  This is a query for collections and data objects
+            // with a trailing wildcard.
+
+            // First get collections
+            query = fmt::format(
+                "select COLL_NAME where COLL_NAME like '{}%' and COLL_NAME not like '{}%/%'",
+                full_path.string(),
+                full_path.string());
+            std::cout << query << std::endl;
+            for (auto&& i : irods::query<RcComm>(thing.get(), query)) {
+                found_objects = true;
+                ptree object;
+                std::string key = (i[0].size() > base_length ? i[0].substr(base_length) : "");
+                if (key.rfind("/", 0) == 0) {
+                    key = key.substr(1);
+                }
+                key += "/";
+                object.put("Prefix", key);
+                document.add_child("ListBucketResult.CommonPrefixes", object);
+            }
+
+            // Now get data objects 
+            query = fmt::format(
+                "select COLL_NAME, DATA_NAME, DATA_OWNER_NAME, DATA_SIZE, DATA_MODIFY_TIME where COLL_NAME = '{}'"
+                " and DATA_NAME like '{}%'",
+                full_path.parent_path().string(),
+                full_path.object_name().string());
```

Comment:
`.c_str()`

Reply:
Same as before.  This is a path object not a std::string.

Reply:
done